### DISCUSSION TITLE
makeGeneAnnotationTrack display more genes?

### DISCUSSION THREAD

**Discussion by mattjvincent** [genome-spy/genome-spy#73] [Q&A] [answered]
Is there a simple way to get more tracks to be displayed?  Maybe 5 instead of 3?

From the following: https://observablehq.com/@tuner/annotation-tracks?collection=@tuner/genomespy

```
   // Display a maximum of three (now 5) lanes to conserve space.
    {
      type: "filter",
      expr: "datum._lane < 5"
    }
```

I changed 3 to 5, but didn't make any difference.  Is there something else I need to do?

Thanks!

**Comment by tuner** [comment, marked as answer]
There's also `domain: [0, 3],` about ten lines later. Try to change that too.

You can also remove it because setting an explicit domain is not necessary, but it's recommended especially with large data sets. Otherwise it needs to be extracted from the data.

**Reply by mattjvincent** [reply]
Ohhhh... I missed that! Thank you!

**Comment by mattjvincent** [comment]
Just as an aside, the pileup transformation does a great job of compacting the data into lanes.  If you remove the `domain: [0, 3]` and also increase the the size of the gene, you can view more at a time.

```
            {
                "type": "formula",
                "expr": "datum._start - (datum.length/2)",
                "as": "_newstart"
            },
            {
                "type": "formula",
                "expr": "datum._start + (datum.length * 1.5)",
                "as": "_newend"
            },
```
...
```
            {
                "type": "collect",
                "sort": {
                    "field": ["_newstart"]
                }
            },
            {
                "type": "pileup",
                "start": "_newstart",
                "end": "_newend",
                "as": "_lane",
                "preference": "strand",
                "preferredOrder": ["-", "+"]
            }
```
